### PR TITLE
[IMP] base, hr: add phone in user preferences

### PR DESCRIPTION
**Pull request by ibo-odoo**
**PURPOSE:**
Enable users to easily update their phone number even if the "Contacts" app is not available.

**SPECIFICATIONS:**
Adding the phone field below the email field in preferences.

Task-4247284

### CODE REVIEW

File: addons/hr/tests/test_self_user_access.py [hunk 1]
```diff
@@ -209,10 +209,9 @@ def testWriteOtherUserPreferencesEmployee(self):
                 self.hubert.with_user(self.richard).write(v)
 
     def testWriteSelfPhoneEmployee(self):
-        # phone is a related from res.partner (from base) but added in SELF_READABLE_FIELDS
         self.env['ir.config_parameter'].set_param('hr.hr_employee_self_edit', False)
         with self.assertRaises(AccessError):
-            self.richard.with_user(self.richard).write({'phone': '2154545'})
+            self.richard.with_user(self.richard).write({'work_phone': '2154545'})
```
Comment (mrsr-odoo) [inline addons/hr/tests/test_self_user_access.py:214]: Hello @ibo-odoo  :wave: ,
It seems this test was to check the phone field, and should raise an access error from orm.
But here using `work_phone` raises a different access error from res.user(inherit in hr). Something like restricting hr fields
And this type of access error is already handled in `testWriteSelfUserEmployeeSettingFalse`. 

Maybe removing this test is better.

Thanks

Comment (ibo-odoo) [reply addons/hr/tests/test_self_user_access.py:214]: I think this is not the case that `this type of access error is already handled in testWriteSelfUserEmployeeSettingFalse`, the test case is specific to the `hr_employee_self_edit` functionality. 

However, as we discussed, earlier the phone field existed in the res_users model in the HR module, and it was later replaced by the employee_phone field. After that, this field was removed as well. It's possible that this test case is no longer needed. Therefore, removing it.
(Leaving this conversation open for feedback from other validators.)

File: addons/hr/tests/test_self_user_access.py [hunk 2]
```diff
@@ -209,12 +209,6 @@ def testWriteOtherUserPreferencesEmployee(self):
             with self.assertRaises(AccessError):
                 self.hubert.with_user(self.richard).write(v)
 
-    def testWriteSelfPhoneEmployee(self):
-        # phone is a related from res.partner (from base) but added in SELF_READABLE_FIELDS
-        self.env['ir.config_parameter'].set_param('hr.hr_employee_self_edit', False)
```
Comment (tde-banana-odoo) [inline addons/hr/tests/test_self_user_access.py:214]: Shouldn't we update ``testWriteOtherUserPreferencesEmployee`` hereabove to show it is now editable ?

File: odoo/addons/base/tests/test_res_users.py [hunk 3]
```diff
@@ -157,6 +157,18 @@ def test_change_user_company(self):
             "the company_id of the partner_id shall be updated"
         )
 
+    def test_user_phone(self):
```
Comment (tde-banana-odoo) [inline odoo/addons/base/tests/test_res_users.py:160]: To make it a bit more generic ?
```suggestion
    def test_user_self_update(self):
```

Comment (tde-banana-odoo) [inline odoo/addons/base/tests/test_res_users.py:160]: Beware alphabetical ordering ! That way tests execution matches tests definition.

**Review by tde-banana-odoo — COMMENTED** [review]
Zboing, thansk for the pr :) some small comments while going through :)

File: addons/hr/tests/test_self_user_access.py (hunk 2 above)
Comment (ibo-odoo) [reply addons/hr/tests/test_self_user_access.py:214]: @tde-banana-odoo Shouldn't it be better to update `testWriteSelfUserPreferencesEmployee`, as it tests whether the user can update `their` non-employee fields without raising an error? (If I misunderstood, please correct me.)

Comment (tde-banana-odoo) [reply addons/hr/tests/test_self_user_access.py:214]: Yes indeed ^^

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**